### PR TITLE
[Arm64] Clean up genPutArgStk

### PR DESCRIPTION
**Pull request by sdmaclea**
Use ldp/stp in more case
Simplify code by eliminating most addrReg conflict cases

### CODE REVIEW

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib @dotnet/jit-contrib PTAL

File: src/jit/codegenarmarch.cpp [hunk 1]
```diff
@@ -284,15 +284,21 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 genConsumeAddress(addrNode);
                 addrReg = addrNode->gtRegNum;
 
+                // If addrReg equal to loReg, swap(loReg, hiReg)
+                // This reduces code complexity by only supporting one addrReg overwrite case
+                if (loReg == addrReg)
+                {
+                    loReg = hiReg;
```
Comment (briansull) [inline src/jit/codegenarmarch.cpp]: Add a comment:
`
// Swap loReg and hiReg`

Comment (briansull) [inline src/jit/codegenarmarch.cpp]: I don't think that you can do it this way.

I there is a conflict you have to reverse the two loads, not swap the registers used.
Loading the high part (offet+8) into the loReg isn't going to work right

I think with this change you'll end up with the wrong values in loReg and hiReg.

Comment (sdmaclea) [inline src/jit/codegenarmarch.cpp]: loReg & hiReg are just temporaries assigned in line 215-216 above.  At this point I do not think either have been used.

Comment (briansull) [inline src/jit/codegenarmarch.cpp]: It will generate an incorrect store sequence, when loReg and hiReg don't contain the lo and hi parts respectively.

**Comment by sdmaclea** [comment]
I don't follow.  By swapping them I only switched the numeric value for the two temporary registers.  
The code still loads and store the low part into loReg and the high part into hiReg.  The code is passing on [Arm64/Unix].

Is there a numeric ordering requirement for the reg numbers loReg and hiReg?.

**Comment by briansull** [comment]
I'm sorry, you can swap the registers number as you have done.
I was confused abut the exact sequence, but tes you can swap these values before issuing any loads or stores.

**Pull request merged**